src/rpc/blockchain.cpp

```diff
@@ -1848,26 +1885,34 @@ static UniValue getblockstats(const JSONRPCRequest& request)
 
             // New feerate uses satoshis per virtual byte instead of per serialized byte
             CAmount feerate = weight ? (txfee * WITNESS_SCALE_FACTOR) / weight : 0;
-            if (do_medianfeerate) {
-                feerate_array.push_back(feerate);
+            if (do_feerate_percentiles) {
+                feerate_array.emplace_back(std::make_pair(feerate, weight));
```

Comment:
With `emplace_back`, I think you should be able to get rid of `make_pair` - `std::pair` has a constructor with `first` and `second`, so that should be called then.

Reply:
So use `std::pair<CAmount, int64_t>(feerate, weight)` instead? Is there a reason the constructor is preferred?

Reply:
Unless there is compelling reason to change the code, you should keep the it as is for now.

Reply:
No, I meant to just use `freerate_array.emplace_back(feerate, weight)`.  I haven't tested it, but according to my understanding it should work - and just be simpler in the code.  Whether that is a compelling reason or not, I don't know.

Reply:
Agree with @domob1812 that
```C++
feerate_array.emplace_back(feerate, weight);
```
looks cleaner (and yes, it works).